src/FSharp.Data.GraphQL.Client.DesignTime/ProvidedTypesHelper.fs

```diff
@@ -340,71 +290,87 @@ module internal ProvidedOperation =
                             let responseJson = Tracer.runAndMeasureExecutionTime "Parsed a GraphQL response to a JsonValue" (fun _ -> JsonValue.Parse response)
                             // If the user does not provide a context, we should dispose the default one after running the query
                             if isDefaultContext then (context :> IDisposable).Dispose()
-                            return OperationResultBase(responseJson, %%operationFieldsExpr, operationTypeName)
-                        } @@>
-                let mdef = ProvidedMethod("AsyncRun", mprm, Types.makeAsync rtdef, invoker)
-                mdef.AddXmlDoc("Executes the operation asynchronously on the server and fetch its results.")
-                mdef
+                            OperationResultBase(responseJson, %%operationFieldsExpr, operationTypeName) @@>
+                    let mprm = varprm |> List.map (fun (name, t) -> ProvidedParameter(name, t))
+                    let mdef = ProvidedMethod("Run", mprm, rtdef, invoker)
+                    mdef.AddXmlDoc("Executes the operation on the server and fetch its results.")
+                    upcast mdef)
+            let arundefs : MemberInfo list = 
+                let operationName = Option.toObj operationDefinition.Name
+                mprm |> List.map (fun varprm ->
+                    // We rebuild our variables by taking the name and type of each parameter (and removing the context parameter)
+                    let variables = varprm |> List.filter (fun (name, _) -> name <> "runtimeContext")
+                    let invoker (args : Expr list) =
+                        // First arg is the operation instance, second should be the context, if the overload has one
+                        // We determine it by calculating the difference in length of variables and arguments
+                        let args, isDefaultContext, context = 
+                            if args.Length - variables.Length = 2
+                            then List.skip 2 args, false, args.[1]
+                            else args.Tail, true, defaultContextExpr
+                        let variables = varExprMapper variables args // Need to remove first arg (opration instance)
+                        <@@ let context = %%context : GraphQLProviderRuntimeContext
+                            let request =
+                                { ServerUrl = context.ServerUrl
+                                  HttpHeaders = context.HttpHeaders
+                                  OperationName = Option.ofObj operationName
+                                  Query = actualQuery
+                                  Variables = %%variables }
+                            async {
+                                let! response = 
+                                    if shouldUseMultipartRequest
```

Comment:
`shouldUseMultipartRequest` should be determined for each operation type and not globally. For every operation type we generate we should be checking the inputs recursively to see if they are the Upload type (or lists of Upload type). We can determine this once at design time for each operation type and cache the result.

Reply:
Done.